modules/client/src/main/scala/ch/epfl/bluebrain/nexus/admin/client/AdminClient.scala

```diff
@@ -0,0 +1,104 @@
+package ch.epfl.bluebrain.nexus.admin.client
+
+import akka.http.scaladsl.client.RequestBuilding.Get
+import akka.http.scaladsl.model.Uri._
+import akka.http.scaladsl.model.headers.OAuth2BearerToken
+import akka.http.scaladsl.model.{HttpResponse, StatusCodes}
+import akka.stream.Materializer
+import ch.epfl.bluebrain.nexus.admin.client.config.AdminConfig
+import ch.epfl.bluebrain.nexus.admin.client.types.Project
+import ch.epfl.bluebrain.nexus.admin.refined.project.ProjectReference
+import ch.epfl.bluebrain.nexus.commons.http.HttpClient.UntypedHttpClient
+import ch.epfl.bluebrain.nexus.commons.http.JsonLdCirceSupport.unmarshaller
+import ch.epfl.bluebrain.nexus.commons.http.{HttpClient, UnexpectedUnsuccessfulHttpResponse}
+import ch.epfl.bluebrain.nexus.commons.iam.acls.FullAccessControlList
+import ch.epfl.bluebrain.nexus.commons.iam.identity.Identity
+import ch.epfl.bluebrain.nexus.commons.iam.io.serialization.JsonLdSerialization
+import ch.epfl.bluebrain.nexus.commons.types.HttpRejection.UnauthorizedAccess
+import io.circe.Decoder
+import io.circe.generic.auto._
+import journal.Logger
+
+import scala.concurrent.{ExecutionContext, Future}
+
+/**
+  * Admin client contract
+  *
+  * @tparam F the monadic effect type
+  */
+trait AdminClient[F[_]] {
+
+  /**
+    * Retrieves a [[Project]] resource instance.
+    *
+    * @param name        the project name
+    * @param credentials the optionally provided [[OAuth2BearerToken]]
+    */
+  def getProject(name: ProjectReference, credentials: Option[OAuth2BearerToken]): F[Project]
```

Comment:
I think the credentials can be implicit since it is the common way we expose them elsewhere
```scala
def getProject(name: ProjectReference)(implicit cred: Option[OAuth2BearerToken])
```